### PR TITLE
Using C# 9.0 features

### PR DESCRIPTION
**Pull request by jbogard**

### CODE REVIEW

File: src/MediatR/Unit.cs [hunk 1]
```diff
@@ -29,10 +29,7 @@ public struct Unit : IEquatable<Unit>, IComparable<Unit>, IComparable
         ///  - Zero: This object is equal to <paramref name="other" />.
         ///  - Greater than zero: This object is greater than <paramref name="other" />.
         /// </returns>
-        public int CompareTo(Unit other)
-        {
-            return 0;
-        }
+        public int CompareTo(Unit other) => 0;
```
Comment (lilasquared) [inline src/MediatR/Unit.cs:32]: Is there a change to editorconfig you can do to make this preferred?

Comment (jbogard) [reply src/MediatR/Unit.cs:32]: Probably, but it'd be R#-only

File: src/MediatR/Internal/RequestHandlerWrapper.cs [hunk 2]
```diff
@@ -48,7 +48,7 @@ internal class RequestHandlerWrapperImpl<TRequest, TResponse> : RequestHandlerWr
             return Handle((IRequest<TResponse>)request, cancellationToken, serviceFactory)
                 .ContinueWith(t =>
                 {
-                    if (t.IsFaulted)
+                    if (t.IsFaulted && t.Exception?.InnerException is not null)
```
Comment (no1melman) [inline src/MediatR/Internal/RequestHandlerWrapper.cs:51]: Would this break functionality - as in, does `ExceptionDispatchInfo` still throw even if the innerEx is null?

Comment (jbogard) [reply src/MediatR/Internal/RequestHandlerWrapper.cs:51]: Yes it does, https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.capture?view=netcore-3.1#exceptions

**Pull request merged**